heron/schedulers/src/java/org/apache/heron/scheduler/kubernetes/V1Controller.java

```diff
@@ -386,7 +386,7 @@ private V1StatefulSet createStatefulSet(Resource containerResource, int numberOf
     statefulSetSpec.selector(selector);
 
     // create a pod template
-    final V1PodTemplateSpec podTemplateSpec = loadPodFromTemplate();
+    final V1PodTemplateSpec podTemplateSpec = new V1PodTemplateSpec();
```

Comment:
Hmm. Maybe it is because `loadPodFromTemplate()` throws an exception?

Reply:
Hi @nwangtw, thank you! I have spun up a build/test Ubuntu18.04 Docker container on my machine and I get the following output with:

```bash
INFO: Elapsed time: 198.476s, Critical Path: 9.83s
INFO: 855 processes: 191 internal, 664 local.
INFO: Build completed successfully, 855 total actions
PASSED: //heron/stmgr/tests/cpp/server:stmgr_unittest (see /root/.cache/bazel/_bazel_root/f4ab758bd53020512013f7dfa13b6902/execroot/org_apache_heron/bazel-out/k8-fastbuild/testlogs/heron/stmgr/tests/cpp/server/stmgr_unittest/test.log)
INFO: From Testing //heron/stmgr/tests/cpp/server:stmgr_unittest
==================== Test output for //heron/stmgr/tests/cpp/server:stmgr_unittest:
Current working directory (to find stmgr logs) /root/.cache/bazel/_bazel_root/f4ab758bd53020512013f7dfa13b6902/execroot/org_apache_heron/bazel-out/k8-fastbuild/bin/heron/stmgr/tests/cpp/server/stmgr_unittest.runfiles/org_apache_heron
Using config file heron/config/src/yaml/conf/test/test_heron_internals.yaml
[==========] Running 11 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 11 tests from StMgr
[ RUN      ] StMgr.test_pplan_decode
[       OK ] StMgr.test_pplan_decode (2011 ms)
[ RUN      ] StMgr.test_tuple_route
[       OK ] StMgr.test_tuple_route (2009 ms)
[ RUN      ] StMgr.test_custom_grouping_route
[       OK ] StMgr.test_custom_grouping_route (2008 ms)
[ RUN      ] StMgr.test_back_pressure_instance
[       OK ] StMgr.test_back_pressure_instance (4010 ms)
[ RUN      ] StMgr.test_spout_death_under_backpressure
[       OK ] StMgr.test_spout_death_under_backpressure (6149 ms)
[ RUN      ] StMgr.test_back_pressure_stmgr
[       OK ] StMgr.test_back_pressure_stmgr (5004 ms)
[ RUN      ] StMgr.test_back_pressure_stmgr_reconnect
[       OK ] StMgr.test_back_pressure_stmgr_reconnect (5033 ms)
[ RUN      ] StMgr.test_tmanager_restart_on_new_address
[       OK ] StMgr.test_tmanager_restart_on_new_address (5014 ms)
[ RUN      ] StMgr.test_tmanager_restart_on_same_address
[       OK ] StMgr.test_tmanager_restart_on_same_address (4016 ms)
[ RUN      ] StMgr.test_metricsmgr_reconnect[       OK ] StMgr.test_metricsmgr_reconnect (3005 ms)
[ RUN      ] StMgr.test_PatchPhysicalPlanWithHydratedTopology
[       OK ] StMgr.test_PatchPhysicalPlanWithHydratedTopology (0 ms)
[----------] 11 tests from StMgr (38260 ms total)
[----------] Global test environment tear-down
[==========] 11 tests from 1 test case ran. (38260 ms total)
[  PASSED  ] 11 tests.
================================================================================
//heron/stmgr/tests/cpp/server:stmgr_unittest                   (cached) PASSED in 38.3s

INFO: Build completed successfully, 855 total actions
```

Re-running with the whole suite yields the following:

```bash
//heron/tmanager/tests/cpp/server:stateful_restorer_unittest (1/3 cached) FLAKY, failed in 2 out of 3 in 5.1s
  Stats over 3 runs: max = 5.1s, min = 0.0s, avg = 1.7s, dev = 2.4s
```

What I can discern from the test logs is that all tests inclusive of `test_custom_grouping_route` pass, which indicates a timeout on `test_back_pressure_instance`:
https://github.com/apache/incubator-heron/blob/396f2b848da0f56dcfda1d917928358133851cf5/heron/stmgr/tests/cpp/server/stmgr_unittest.cpp#L960

I have refactored the code a little to make it more readable on the `dev` branch but that should not be causing this test to pass locally. I shall clean up the Git commit-tree and merge the changes on `dev` into the main feature branch to see if it now passes on the Travis CI build.

Reply:
The following timer warnings are being output, full log is below:

```bash
  WARNING: //heron/api/tests/scala:api-scala-test: Test execution time (0.9s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
  WARNING: //heron/stmgr/tests/cpp/server:checkpoint-gateway_unittest: Test execution time (1.2s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
  WARNING: //heron/stmgr/tests/cpp/server:stateful-restorer_unittest: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
  WARNING: //heron/stmgr/tests/cpp/util:neighbour_calculator_unittest: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
```

<details><summary>Test ouput</summary>
<p>

```bash
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=100
INFO: Reading rc options for 'test' from /heron/tools/bazel.rc:
  Inherited 'build' options: --genrule_strategy=standalone --host_force_python=PY3 --ignore_unsupported_sandboxing --spawn_strategy=standalone --workspace_status_command scripts/release/status.sh
INFO: Reading rc options for 'test' from /root/.bazelrc:
  Inherited 'build' options: --verbose_failures --host_force_python=PY3 --spawn_strategy=standalone --genrule_strategy=standalone --local_ram_resources=4096 --local_cpu_resources=2 --announce_rc
INFO: Reading rc options for 'test' from /root/.bazelrc:
  'test' options: --test_strategy=standalone
INFO: Found applicable config definition build:ubuntu in file /heron/tools/bazel.rc: --experimental_action_listener=tools/java:compile_java --experimental_action_listener=tools/cpp:compile_cpp --experimental_action_listener=tools/python:compile_python --genrule_strategy=standalone --ignore_unsupported_sandboxing --linkopt -lm --linkopt -lpthread --linkopt -lrt --spawn_strategy=standalone --workspace_status_command scripts/release/status.sh --copt=-O3
WARNING: /heron/heron/healthmgr/tests/java/BUILD:52:13: in srcs attribute of java_library rule //heron/healthmgr/tests/java:healthmgr-tests: please do not import '//heron/healthmgr/src/java:org/apache/heron/healthmgr/HealthManager.java' directly. You should either move the file to this package or depend on an appropriate rule there. Since this rule was created by the macro 'java_library', the error might have been caused by the macro implementation
INFO: Analyzed 612 targets (0 packages loaded, 0 targets configured).
INFO: Found 379 targets and 233 test targets...
INFO: Elapsed time: 148.802s, Critical Path: 49.52s
INFO: 234 processes: 1 internal, 233 local.
INFO: Build completed successfully, 234 total actions
//heron/api/tests/cpp:serialization_unittest                             PASSED in 0.0s
//heron/api/tests/java:BaseWindowedBoltTest                              PASSED in 0.3s
//heron/api/tests/java:ConfigTest                                        PASSED in 0.4s
//heron/api/tests/java:CountStatAndMetricTest                            PASSED in 0.2s
//heron/api/tests/java:GeneralReduceByKeyAndWindowOperatorTest           PASSED in 0.4s
//heron/api/tests/java:HeronSubmitterTest                                PASSED in 2.0s
//heron/api/tests/java:JoinOperatorTest                                  PASSED in 0.6s
//heron/api/tests/java:KVStreamletShadowTest                             PASSED in 0.4s
//heron/api/tests/java:KeyByOperatorTest                                 PASSED in 0.5s
//heron/api/tests/java:LatencyStatAndMetricTest                          PASSED in 0.4s
//heron/api/tests/java:ReduceByKeyAndWindowOperatorTest                  PASSED in 0.5s
//heron/api/tests/java:StreamletImplTest                                 PASSED in 0.4s
//heron/api/tests/java:StreamletShadowTest                               PASSED in 0.4s
//heron/api/tests/java:StreamletUtilsTest                                PASSED in 0.3s
//heron/api/tests/java:UtilsTest                                         PASSED in 0.3s
//heron/api/tests/java:WaterMarkEventGeneratorTest                       PASSED in 0.3s
//heron/api/tests/java:WindowManagerTest                                 PASSED in 0.3s
//heron/api/tests/java:WindowedBoltExecutorTest                          PASSED in 0.5s
//heron/api/tests/scala:api-scala-test                                   PASSED in 0.9s
  WARNING: //heron/api/tests/scala:api-scala-test: Test execution time (0.9s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//heron/ckptmgr/tests/java:CheckpointManagerServerTest                   PASSED in 0.7s
//heron/common/tests/cpp/basics:fileutils_unittest                       PASSED in 0.0s
//heron/common/tests/cpp/basics:rid_unittest                             PASSED in 0.0s
//heron/common/tests/cpp/basics:strutils_unittest                        PASSED in 0.0s
//heron/common/tests/cpp/basics:utils_unittest                           PASSED in 0.0s
//heron/common/tests/cpp/config:topology-config-helper_unittest          PASSED in 0.0s
//heron/common/tests/cpp/errors:errors_unittest                          PASSED in 0.0s
//heron/common/tests/cpp/errors:module_unittest                          PASSED in 0.0s
//heron/common/tests/cpp/errors:syserrs_unittest                         PASSED in 0.0s
//heron/common/tests/cpp/metrics:count-metric_unittest                   PASSED in 0.0s
//heron/common/tests/cpp/metrics:mean-metric_unittest                    PASSED in 0.0s
//heron/common/tests/cpp/metrics:multi-count-metric_unittest             PASSED in 0.0s
//heron/common/tests/cpp/metrics:multi-mean-metric_unittest              PASSED in 0.0s
//heron/common/tests/cpp/metrics:time-spent-metric_unittest              PASSED in 1.3s
//heron/common/tests/cpp/network:http_unittest                           PASSED in 0.1s
//heron/common/tests/cpp/network:order_unittest                          PASSED in 0.0s
//heron/common/tests/cpp/network:packet_unittest                         PASSED in 0.0s
//heron/common/tests/cpp/network:piper_unittest                          PASSED in 2.0s
//heron/common/tests/cpp/network:rate_limit_unittest                     PASSED in 4.0s
//heron/common/tests/cpp/network:switch_unittest                         PASSED in 0.2s
//heron/common/tests/cpp/threads:spcountdownlatch_unittest               PASSED in 2.0s
//heron/common/tests/java:ByteAmountTest                                 PASSED in 0.3s
//heron/common/tests/java:CommunicatorTest                               PASSED in 0.3s
//heron/common/tests/java:ConfigReaderTest                               PASSED in 0.5s
//heron/common/tests/java:EchoTest                                       PASSED in 0.8s
//heron/common/tests/java:FileUtilsTest                                  PASSED in 1.0s
//heron/common/tests/java:HeronServerTest                                PASSED in 1.6s
//heron/common/tests/java:PackageTypeTest                                PASSED in 0.3s
//heron/common/tests/java:SysUtilsTest                                   PASSED in 5.1s
//heron/common/tests/java:SystemConfigTest                               PASSED in 0.3s
//heron/common/tests/java:TopologyUtilsTest                              PASSED in 0.4s
//heron/common/tests/java:WakeableLooperTest                             PASSED in 1.3s
//heron/common/tests/python/pex_loader:pex_loader_unittest               PASSED in 0.7s
//heron/downloaders/tests/java:DLDownloaderTest                          PASSED in 0.8s
//heron/downloaders/tests/java:ExtractorTests                            PASSED in 0.3s
//heron/downloaders/tests/java:RegistryTest                              PASSED in 0.5s
//heron/executor/tests/python:executor_unittest                          PASSED in 1.1s
//heron/healthmgr/tests/java:BackPressureDetectorTest                    PASSED in 0.5s
//heron/healthmgr/tests/java:BackPressureSensorTest                      PASSED in 0.6s
//heron/healthmgr/tests/java:BufferSizeSensorTest                        PASSED in 0.6s
//heron/healthmgr/tests/java:DataSkewDiagnoserTest                       PASSED in 0.6s
//heron/healthmgr/tests/java:ExecuteCountSensorTest                      PASSED in 0.6s
//heron/healthmgr/tests/java:GrowingWaitQueueDetectorTest                PASSED in 0.5s
//heron/healthmgr/tests/java:HealthManagerTest                           PASSED in 0.8s
//heron/healthmgr/tests/java:HealthPolicyConfigReaderTest                PASSED in 0.5s
//heron/healthmgr/tests/java:LargeWaitQueueDetectorTest                  PASSED in 0.5s
//heron/healthmgr/tests/java:MetricsCacheMetricsProviderTest             PASSED in 0.6s
//heron/healthmgr/tests/java:PackingPlanProviderTest                     PASSED in 0.6s
//heron/healthmgr/tests/java:ProcessingRateSkewDetectorTest              PASSED in 0.6s
//heron/healthmgr/tests/java:ScaleUpResolverTest                         PASSED in 0.7s
//heron/healthmgr/tests/java:SlowInstanceDiagnoserTest                   PASSED in 0.7s
//heron/healthmgr/tests/java:UnderProvisioningDiagnoserTest              PASSED in 0.6s
//heron/healthmgr/tests/java:WaitQueueSkewDetectorTest                   PASSED in 0.7s
//heron/instance/tests/java:ActivateDeactivateTest                       PASSED in 0.6s
//heron/instance/tests/java:BoltInstanceTest                             PASSED in 0.4s
//heron/instance/tests/java:BoltStatefulInstanceTest                     PASSED in 2.6s
//heron/instance/tests/java:ConnectTest                                  PASSED in 0.7s
//heron/instance/tests/java:CustomGroupingTest                           PASSED in 0.6s
//heron/instance/tests/java:EmitDirectBoltTest                           PASSED in 0.6s
//heron/instance/tests/java:EmitDirectSpoutTest                          PASSED in 0.7s
//heron/instance/tests/java:GlobalMetricsTest                            PASSED in 0.3s
//heron/instance/tests/java:HandleReadTest                               PASSED in 0.8s
//heron/instance/tests/java:HandleWriteTest                              PASSED in 5.5s
//heron/instance/tests/java:MultiAssignableMetricTest                    PASSED in 0.3s
//heron/instance/tests/java:SpoutInstanceTest                            PASSED in 2.6s
//heron/instance/tests/java:SpoutStatefulInstanceTest                    PASSED in 2.5s
//heron/instance/tests/python/network:event_looper_unittest              PASSED in 3.3s
//heron/instance/tests/python/network:gateway_looper_unittest            PASSED in 10.8s
//heron/instance/tests/python/network:heron_client_unittest              PASSED in 1.0s
//heron/instance/tests/python/network:metricsmgr_client_unittest         PASSED in 1.4s
//heron/instance/tests/python/network:protocol_unittest                  PASSED in 0.9s
//heron/instance/tests/python/network:st_stmgrcli_unittest               PASSED in 0.9s
//heron/instance/tests/python/utils:communicator_unittest                PASSED in 0.9s
//heron/instance/tests/python/utils:custom_grouping_unittest             PASSED in 0.9s
//heron/instance/tests/python/utils:global_metrics_unittest              PASSED in 0.9s
//heron/instance/tests/python/utils:log_unittest                         PASSED in 0.8s
//heron/instance/tests/python/utils:metrics_helper_unittest              PASSED in 0.9s
//heron/instance/tests/python/utils:outgoing_tuple_helper_unittest       PASSED in 0.9s
//heron/instance/tests/python/utils:pplan_helper_unittest                PASSED in 0.9s
//heron/instance/tests/python/utils:py_metrics_unittest                  PASSED in 0.9s
//heron/instance/tests/python/utils:topology_context_impl_unittest       PASSED in 1.1s
//heron/instance/tests/python/utils:tuple_helper_unittest                PASSED in 1.4s
//heron/io/dlog/tests/java:DLInputStreamTest                             PASSED in 0.5s
//heron/io/dlog/tests/java:DLOutputStreamTest                            PASSED in 0.6s
//heron/metricscachemgr/tests/java:CacheCoreTest                         PASSED in 0.4s
//heron/metricscachemgr/tests/java:MetricsCacheQueryUtilsTest            PASSED in 0.4s
//heron/metricscachemgr/tests/java:MetricsCacheTest                      PASSED in 0.4s
//heron/metricsmgr/tests/java:FileSinkTest                               PASSED in 0.4s
//heron/metricsmgr/tests/java:HandleTManagerLocationTest                 PASSED in 0.5s
//heron/metricsmgr/tests/java:MetricsCacheSinkTest                       PASSED in 9.5s
//heron/metricsmgr/tests/java:MetricsManagerServerTest                   PASSED in 0.6s
//heron/metricsmgr/tests/java:MetricsUtilTests                           PASSED in 0.3s
//heron/metricsmgr/tests/java:PrometheusSinkTests                        PASSED in 0.4s
//heron/metricsmgr/tests/java:SinkExecutorTest                           PASSED in 0.5s
//heron/metricsmgr/tests/java:TManagerSinkTest                           PASSED in 9.5s
//heron/metricsmgr/tests/java:WebSinkTest                                PASSED in 0.4s
//heron/packing/tests/java:FirstFitDecreasingPackingTest                 PASSED in 0.8s
//heron/packing/tests/java:PackingPlanBuilderTest                        PASSED in 0.5s
//heron/packing/tests/java:PackingUtilsTest                              PASSED in 0.4s
//heron/packing/tests/java:ResourceCompliantRRPackingTest                PASSED in 0.7s
//heron/packing/tests/java:RoundRobinPackingTest                         PASSED in 0.6s
//heron/packing/tests/java:ScorerTest                                    PASSED in 0.2s
//heron/scheduler-core/tests/java:HttpServiceSchedulerClientTest         PASSED in 1.1s
//heron/scheduler-core/tests/java:JsonFormatterUtilsTest                 PASSED in 0.4s
//heron/scheduler-core/tests/java:LaunchRunnerTest                       PASSED in 1.1s
//heron/scheduler-core/tests/java:LauncherUtilsTest                      PASSED in 2.0s
//heron/scheduler-core/tests/java:LibrarySchedulerClientTest             PASSED in 0.4s
//heron/scheduler-core/tests/java:RuntimeManagerMainTest                 PASSED in 2.5s
//heron/scheduler-core/tests/java:RuntimeManagerRunnerTest               PASSED in 2.3s
//heron/scheduler-core/tests/java:SchedulerClientFactoryTest             PASSED in 1.2s
//heron/scheduler-core/tests/java:SchedulerMainTest                      PASSED in 3.3s
//heron/scheduler-core/tests/java:SchedulerServerTest                    PASSED in 0.4s
//heron/scheduler-core/tests/java:SchedulerUtilsTest                     PASSED in 1.3s
//heron/scheduler-core/tests/java:SubmitDryRunRenderTest                 PASSED in 1.5s
//heron/scheduler-core/tests/java:SubmitterMainTest                      PASSED in 1.5s
//heron/scheduler-core/tests/java:UpdateDryRunRenderTest                 PASSED in 1.7s
//heron/scheduler-core/tests/java:UpdateTopologyManagerTest              PASSED in 12.1s
//heron/schedulers/tests/java:AuroraCLIControllerTest                    PASSED in 0.4s
//heron/schedulers/tests/java:AuroraContextTest                          PASSED in 0.3s
//heron/schedulers/tests/java:AuroraLauncherTest                         PASSED in 0.7s
//heron/schedulers/tests/java:AuroraSchedulerTest                        PASSED in 2.6s
//heron/schedulers/tests/java:HeronExecutorTaskTest                      PASSED in 1.6s
//heron/schedulers/tests/java:HeronMasterDriverTest                      PASSED in 1.7s
//heron/schedulers/tests/java:KubernetesContextTest                      PASSED in 0.3s
//heron/schedulers/tests/java:KubernetesControllerTest                   PASSED in 0.4s
//heron/schedulers/tests/java:KubernetesLauncherTest                     PASSED in 0.9s
//heron/schedulers/tests/java:KubernetesSchedulerTest                    PASSED in 0.7s
//heron/schedulers/tests/java:LaunchableTaskTest                         PASSED in 0.5s
//heron/schedulers/tests/java:LocalLauncherTest                          PASSED in 1.0s
//heron/schedulers/tests/java:LocalSchedulerTest                         PASSED in 0.6s
//heron/schedulers/tests/java:MarathonControllerTest                     PASSED in 1.3s
//heron/schedulers/tests/java:MarathonLauncherTest                       PASSED in 0.7s
//heron/schedulers/tests/java:MarathonSchedulerTest                      PASSED in 0.5s
//heron/schedulers/tests/java:MesosFrameworkTest                         PASSED in 0.6s
//heron/schedulers/tests/java:MesosLauncherTest                          PASSED in 0.4s
//heron/schedulers/tests/java:MesosSchedulerTest                         PASSED in 0.9s
//heron/schedulers/tests/java:NomadSchedulerTest                         PASSED in 2.5s
//heron/schedulers/tests/java:SlurmControllerTest                        PASSED in 1.3s
//heron/schedulers/tests/java:SlurmLauncherTest                          PASSED in 0.9s
//heron/schedulers/tests/java:SlurmSchedulerTest                         PASSED in 1.4s
//heron/schedulers/tests/java:TaskResourcesTest                          PASSED in 0.3s
//heron/schedulers/tests/java:TaskUtilsTest                              PASSED in 0.3s
//heron/schedulers/tests/java:V1ControllerTest                           PASSED in 1.4s
//heron/schedulers/tests/java:VolumesTests                               PASSED in 0.3s
//heron/schedulers/tests/java:YarnLauncherTest                           PASSED in 0.6s
//heron/schedulers/tests/java:YarnSchedulerTest                          PASSED in 0.5s
//heron/simulator/tests/java:AllGroupingTest                             PASSED in 0.3s
//heron/simulator/tests/java:CustomGroupingTest                          PASSED in 0.3s
//heron/simulator/tests/java:FieldsGroupingTest                          PASSED in 0.8s
//heron/simulator/tests/java:InstanceExecutorTest                        PASSED in 0.4s
//heron/simulator/tests/java:LowestGroupingTest                          PASSED in 0.3s
//heron/simulator/tests/java:RotatingMapTest                             PASSED in 0.3s
//heron/simulator/tests/java:ShuffleGroupingTest                         PASSED in 0.3s
//heron/simulator/tests/java:SimulatorTest                               PASSED in 0.4s
//heron/simulator/tests/java:TopologyManagerTest                         PASSED in 0.5s
//heron/simulator/tests/java:TupleCacheTest                              PASSED in 0.4s
//heron/simulator/tests/java:XORManagerTest                              PASSED in 0.4s
//heron/spi/tests/java:ConfigLoaderTest                                  PASSED in 1.2s
//heron/spi/tests/java:ConfigTest                                        PASSED in 0.9s
//heron/spi/tests/java:ContextTest                                       PASSED in 0.3s
//heron/spi/tests/java:ExceptionInfoTest                                 PASSED in 0.2s
//heron/spi/tests/java:KeysTest                                          PASSED in 0.3s
//heron/spi/tests/java:MetricsInfoTest                                   PASSED in 0.3s
//heron/spi/tests/java:MetricsRecordTest                                 PASSED in 0.2s
//heron/spi/tests/java:NetworkUtilsTest                                  PASSED in 1.3s
//heron/spi/tests/java:PackingPlanTest                                   PASSED in 0.3s
//heron/spi/tests/java:ResourceTest                                      PASSED in 0.3s
//heron/spi/tests/java:ShellUtilsTest                                    PASSED in 1.9s
//heron/spi/tests/java:TokenSubTest                                      PASSED in 0.3s
//heron/spi/tests/java:UploaderUtilsTest                                 PASSED in 0.5s
//heron/statefulstorages/tests/java:DlogStorageTest                      PASSED in 2.1s
//heron/statefulstorages/tests/java:HDFSStorageTest                      PASSED in 1.8s
//heron/statefulstorages/tests/java:LocalFileSystemStorageTest           PASSED in 1.1s
//heron/statemgrs/tests/cpp:zk-statemgr_unittest                         PASSED in 0.0s
//heron/statemgrs/tests/java:CuratorStateManagerTest                     PASSED in 0.7s
//heron/statemgrs/tests/java:LocalFileSystemStateManagerTest             PASSED in 1.3s
//heron/statemgrs/tests/java:ZkUtilsTest                                 PASSED in 1.2s
//heron/statemgrs/tests/python:configloader_unittest                     PASSED in 0.9s
//heron/statemgrs/tests/python:statemanagerfactory_unittest              PASSED in 1.4s
//heron/statemgrs/tests/python:zkstatemanager_unittest                   PASSED in 0.8s
//heron/stmgr/tests/cpp/grouping:all-grouping_unittest                   PASSED in 0.0s
//heron/stmgr/tests/cpp/grouping:custom-grouping_unittest                PASSED in 0.0s
//heron/stmgr/tests/cpp/grouping:fields-grouping_unittest                PASSED in 0.0s
//heron/stmgr/tests/cpp/grouping:lowest-grouping_unittest                PASSED in 0.0s
//heron/stmgr/tests/cpp/grouping:shuffle-grouping_unittest               PASSED in 0.0s
//heron/stmgr/tests/cpp/server:checkpoint-gateway_unittest               PASSED in 1.2s
  WARNING: //heron/stmgr/tests/cpp/server:checkpoint-gateway_unittest: Test execution time (1.2s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//heron/stmgr/tests/cpp/server:stateful-restorer_unittest                PASSED in 0.0s
  WARNING: //heron/stmgr/tests/cpp/server:stateful-restorer_unittest: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//heron/stmgr/tests/cpp/server:stmgr_unittest                            PASSED in 40.2s
//heron/stmgr/tests/cpp/util:neighbour_calculator_unittest               PASSED in 0.0s
  WARNING: //heron/stmgr/tests/cpp/util:neighbour_calculator_unittest: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//heron/stmgr/tests/cpp/util:rotating-map_unittest                       PASSED in 0.0s
//heron/stmgr/tests/cpp/util:tuple-cache_unittest                        PASSED in 3.7s
//heron/stmgr/tests/cpp/util:xor-manager_unittest                        PASSED in 4.0s
//heron/tmanager/tests/cpp/server:stateful_checkpointer_unittest         PASSED in 0.0s
//heron/tmanager/tests/cpp/server:stateful_restorer_unittest             PASSED in 5.0s
//heron/tmanager/tests/cpp/server:tcontroller_unittest                   PASSED in 0.0s
//heron/tmanager/tests/cpp/server:tmanager_unittest                      PASSED in 27.0s
//heron/tools/apiserver/tests/java:ConfigUtilsTests                      PASSED in 0.4s
//heron/tools/apiserver/tests/java:TopologyResourceTests                 PASSED in 0.8s
//heron/tools/cli/tests/python:client_command_unittest                   PASSED in 1.0s
//heron/tools/cli/tests/python:opts_unittest                             PASSED in 0.8s
//heron/tools/explorer/tests/python:explorer_unittest                    PASSED in 1.0s
//heron/tools/tracker/tests/python:query_operator_unittest               PASSED in 1.4s
//heron/tools/tracker/tests/python:query_unittest                        PASSED in 1.3s
//heron/tools/tracker/tests/python:topology_unittest                     PASSED in 1.2s
//heron/tools/tracker/tests/python:tracker_unittest                      PASSED in 1.2s
//heron/uploaders/tests/java:DlogUploaderTest                            PASSED in 0.6s
//heron/uploaders/tests/java:GcsUploaderTests                            PASSED in 0.7s
//heron/uploaders/tests/java:HdfsUploaderTest                            PASSED in 0.4s
//heron/uploaders/tests/java:HttpUploaderTest                            PASSED in 0.6s
//heron/uploaders/tests/java:LocalFileSystemConfigTest                   PASSED in 0.3s
//heron/uploaders/tests/java:LocalFileSystemContextTest                  PASSED in 0.3s
//heron/uploaders/tests/java:LocalFileSystemUploaderTest                 PASSED in 0.3s
//heron/uploaders/tests/java:S3UploaderTest                              PASSED in 0.9s
//heron/uploaders/tests/java:ScpUploaderTest                             PASSED in 0.4s

INFO: Build completed successfully, 234 total actions
```

</p>
</details>